src/cmake/compiler.cmake

```diff
@@ -327,11 +327,29 @@ if (NOT USE_SIMD STREQUAL "")
     if (USE_SIMD STREQUAL "0")
         set (SIMD_COMPILE_FLAGS ${SIMD_COMPILE_FLAGS} "-DOIIO_NO_SIMD=1")
     else ()
+        set(_highest_msvc_arch 0)
         string (REPLACE "," ";" SIMD_FEATURE_LIST ${USE_SIMD})
         foreach (feature ${SIMD_FEATURE_LIST})
             message (VERBOSE "SIMD feature: ${feature}")
             if (MSVC OR CMAKE_COMPILER_IS_INTEL)
-                list (APPEND SIMD_COMPILE_FLAGS "/arch:${feature}")
+                if (feature STREQUAL "sse2")
+                    add_compile_definitions(__SSE2__)
```

Comment:
I'm not so sure about directly setting `__SSE2__` and other compile definitions. My understanding is that you should use `-mXXX` on gcc/clang and `/arch:XXX` on MSVS, and that the compilers set the `__BLAH__` symbols.

Reply:
Also, in general, I think we've moved away from directly using `add_compile_definitions` globally (dubious if we want to be safe to be included as a subproject), and instead (as you can see in the surrounding code) appending to SIMD_COMPILE_FLAGS, which will later be used for `proj_add_compile_options()` which is our OIIO-specific way to collect compile options that will be used for our individual libraries and binaries that we are building.

Reply:
There was a bit more context in the Issue for this PR but MSVC will not set the `__SSE*__` defines at all (and there are no `arch` flags for SSE in general).   The only arch flags that exist are for AVX, which oddly enough will set the `__AVX*__` defines on its own.

https://learn.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-170

Reply:
I see. So gcc/clang will set the `__BLAH__` symbols based on the `-m` settings, but with MSVS it's hit or miss? In MS land, how are you expected to test (`#if`, etc) the settings in the code?

OK, so I retract what I said about defining the symbols -- let's do that on MSVS (and icc) to ensure they're set. I'm very sure it's not needed in the gcc/clang compatible compilers.

Then the only suggestion I have is to use our `proj_add_compile_options` instead of setting them globally with `add_compile_definitions` so that we aren't polluting the definitions of any surrounding project.

Reply:
>  but with MSVS it's hit or miss?
Not so much "hit or miss", more than MSVC simply lacks any machinery to target SSE3/SSE4 at all. It only ever has "please target SSE2" (which is implied by mere x64 for all practical purposes these days), or "please target AVX1 or AVX2", for which it does set proper preprocessor defines. For e.g. "I'm compiling for SSE4" case there simply is nothing - the compiler itself will never generate SSE4 instructions, but if you use intrinsics yourself, then they will work, and any runtime checks to guard them (or not do that) is your responsibility.

So realistically people end up doing things like:
- Do I have SSE2? Check if either `__SSE2__` is defined (clang/gcc) or we're simply compiling for x64 (i.e. either `__x86_64__` or `_M_X64` is defined)
- Do I have various AVX variants? Check various AVX defines (`__AVX__`, `__AVX2__`, `__AVX512F__`)
- Do I have SSE3 or SSE4? For gcc/clang check the defines, for MSVC it is "uhh, indicate presence of them in some other way". Which is exactly what this PR is trying to achieve here.

Reply:
Yes, I understand that not all target levels are directly supported in MSVS - only SSE2, AVX, AVX2, AVX512.

What I meant by "hit or miss" is that with MSVS, selecting AVX varieties from the command line defines the `__AVXx__` symbols, but selecting (or defaulting to, on x86_64) SSE does not set any of the `__SSEx__` symbols. Gcc/clang do it uniformly, so I was not aware MSVS did not set the symbols for the SSE levels.